test/extended/networking/whereabouts.go

```diff
@@ -0,0 +1,95 @@
+
+	oc = exutil.NewCLI("whereabouts-e2e")
+
+	f := oc.KubeFramework()
+	podName := "whereabouts-pod-"
+
+	// Whereabouts is already installed in Origin. These tests aims to verify the integrity of the installation.
+
+	g.It("should use whereabouts net-attach-def to limit IP ranges for newly created pods", func() {
+		var err error
+		ns = f.Namespace.Name
+
+		g.By("creating a whereabouts net-attach-def using bridgeCNI")
+		nad_yaml := exutil.FixturePath("testdata", "net-attach-defs", "whereabouts-nad.yml")
+		g.By(fmt.Sprintf("calling oc create -f %s", nad_yaml))
+		err = oc.AsAdmin().Run("create").Args("-f", nad_yaml).Execute()
+		o.Expect(err).NotTo(o.HaveOccurred(), "created net-attach-def")
+
+		g.By("launching pods with annotations to use the net-attach-def")
+		annotation := map[string]string{
+			"k8s.v1.cni.cncf.io/networks": "wa-conf",
+		}
+		// First three pods should come up without issue.
+		testPod := frameworkpod.CreateExecPodOrFail(f.ClientSet, ns, podName, func(pod *v1.Pod) {
+			pod.ObjectMeta.Annotations = annotation
+		})
+		testPod2 := frameworkpod.CreateExecPodOrFail(f.ClientSet, ns, podName, func(pod *v1.Pod) {
+			pod.ObjectMeta.Annotations = annotation
+		})
+		testPod3 := frameworkpod.CreateExecPodOrFail(f.ClientSet, ns, podName, func(pod *v1.Pod) {
+			pod.ObjectMeta.Annotations = annotation
+		})
+		// Fourth pod should not come up.
+		fmt.Println("...creating 4th exec pod, expecting failure due to IPAM")
+		ExecPodSpec := e2epod.NewAgnhostPod(ns, "", nil, nil, nil)
+		ExecPodSpec.ObjectMeta.GenerateName = podName
+		ExecPodSpec.ObjectMeta.Annotations = annotation
+		f.ClientSet.CoreV1().Pods(ns).Create(context.TODO(), ExecPodSpec, metav1.CreateOptions{})
+		time.Sleep(30 * time.Second)
+
+		g.By("checking that additional pods do not come up when range is exhausted")
+		output, err := oc.AsAdmin().Run("get").Args("pods").Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		getPodArr := strings.Split(output, "\n")
+		failedPodCount := 0
+		for _, v := range getPodArr {
+			if strings.Contains(v, "ContainerCreating") {
+				failedPodCount++
+				if failedPodCount > 1 {
+					break
+				}
+			}
+		}
+		o.Expect(failedPodCount).To(o.Equal(1))
+		fmt.Println(output)
+
+		g.By("checking that successfully started pods are within IP range")
+		// pod 1
+		output, err = oc.AsAdmin().Run("exec").Args(testPod.Name, "--", "ip", "a").Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		fmt.Println(output)
+		o.Expect(output).Should(o.ContainSubstring("192.168.2.228"))
+		// pod 2
+		output, err = oc.AsAdmin().Run("exec").Args(testPod2.Name, "--", "ip", "a").Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		fmt.Println(output)
+		o.Expect(output).Should(o.ContainSubstring("192.168.2.229"))
+		// pod 3
+		output, err = oc.AsAdmin().Run("exec").Args(testPod3.Name, "--", "ip", "a").Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		fmt.Println(output)
+		o.Expect(output).Should(o.ContainSubstring("192.168.2.230"))
+
```

Comment:
Something to consider would be adding a check of the "results object" -- that is, the resulting `network-status` annotation on the pod, to see that the proper IP addresses were returned to Multus, and that they show up the status annotation properly.

This would be kind of buttering both sides of the bread, but, might be a nice addition.